presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusClient.java

```diff
@@ -0,0 +1,211 @@
+            String status = (String) tableSupplier.get().get("status");
+            if (status.equals("success")) {
+                List<String> tableNames = (List<String>) tableSupplier.get().get("data");
+                if (tableNames == null) {
+                    return ImmutableSet.of();
+                }
+                return tableNames.stream().collect(Collectors.toSet());
+            }
+            else {
+                //TODO this deals with success|error, there may be warnings to handle
+                return ImmutableSet.of();
+            }
+        }
+        return ImmutableSet.of();
+    }
+
+    public PrometheusTable getTable(String schema, String tableName)
+    {
+        requireNonNull(schema, "schema is null");
+        requireNonNull(tableName, "tableName is null");
+        if (schema.equals("default")) {
+            List<String> tableNames = (List<String>) tableSupplier.get().get("data");
+            if (tableNames == null) {
+                return null;
+            }
+            if (!tableNames.contains(tableName)) {
+                return null;
+            }
+            PrometheusTable table = new PrometheusTable(
+                    tableName,
+                    prometheusColumns);
+            pushedDown.ifPresent(pushedDownPredicate -> table.setPredicate(Optional.of(pushedDownPredicate)));
+            return table;
+        }
+        else {
+            return null;
+        }
+    }
+
+    private static Supplier<Map<String, Object>> metricsSupplier(final JsonCodec<Map<String, Object>> metricsCodec, final URI metadataUri)
+    {
+        return () -> {
+            try {
+                return lookupMetrics(metadataUri, metricsCodec);
+            }
+            catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        };
+    }
+
+    private static Map<String, Object> lookupMetrics(URI metadataUri, JsonCodec<Map<String, Object>> metricsCodec)
+            throws IOException
+    {
+        String json = getHttpResponse(metadataUri).string();
+        Map<String, Object> metrics = metricsCodec.fromJson(json);
+
+        return metrics;
+    }
+
+    protected void setPredicate(Optional<TupleDomain<ColumnHandle>> pushedDown)
+    {
+        this.pushedDown = pushedDown;
+    }
+
+    static ResponseBody getHttpResponse(URI uri)
+            throws IOException
+    {
+        Request.Builder requestBuilder = new Request.Builder();
+        getBearerAuthInfoFromFile().map(bearerToken ->
+                requestBuilder.header("Authorization", "Bearer " + bearerToken));
+        requestBuilder.url(uri.toURL());
+        Request request = requestBuilder.build();
+        Response response = httpClient.newCall(request).execute();
+        if (response.isSuccessful()) {
+            return response.body();
+        }
+        else {
+            throw new PrestoException(REMOTE_TASK_ERROR, "Bad response " + response.code() + response.message());
```

Comment:
REMOTE_TASK_ERROR is dedicated to Presto internal processing. Please define prometheus related errors provider.

Reply:
Done in 486dd9a9